ssl/ssl_lib.c

```diff
@@ -4155,6 +4243,19 @@ void SSL_CTX_free(SSL_CTX *a)
         return;
     REF_ASSERT_ISNT(i < 0);
 
+#ifndef OPENSSL_NO_SSLKEYLOG
+    if (keylog_lock != NULL && CRYPTO_THREAD_write_lock(keylog_lock)) {
```

Comment:
I think we should also test if context we are freeing here has a `->sslkeylog_callback` set here. We should be doing nothing if no `sslkeylog_callback` is set for particular context.

Reply:
If sslkeylog_callback is null here all the implied operations are going to be skipped anyway, as keylog_lock is only non-NULL if sslkeylog_callback is set as well (see setup code in SSL_CTX_new_ex)

Reply:
hmm, ok, thats fair (a bit of an odd use case, but certainly possible).

To remedy this, I think what we need to do is in SSL_CTX_free, we should gate the decrement of the keylog_count on a->sslkeylog_callback being non-null (which may have been what you were alluding to above).  That way, any context that was created prior to the environment variable getting set won't impact the refcount, which I think is what we want here.

I'll fix that